app/commands/convene.tsx

```diff
@@ -0,0 +1,158 @@
+        instance.deactivate();
+        switch (resolution) {
+          case resolutions.restrict:
+            reportUser({
+              reason: ReportReasons.mod,
+              message,
+              extra: "✅ Restricted",
+            });
+            await applyRestriction(message.member!);
+            message.reply(
+              "After a vote by the mods, this member has had restrictions applied to them",
+            );
+            return;
+          case resolutions.kick:
+            reportUser({
+              reason: ReportReasons.mod,
+              message,
+              extra: "✅ Kicked",
+            });
+
+            await kick(message.member!);
+            message.reply(
+              "After a vote by the mods, this member has been kicked from the server to cool off",
+            );
+            return;
+          case resolutions.ban:
+            reportUser({
+              reason: ReportReasons.mod,
+              message,
+              extra: "✅ Banned",
+            });
+
+            await ban(message.member!);
+            message.reply(
+              "After a vote by the mods, this member has been permanently banned",
+            );
+            return;
+          case resolutions.nudge:
+            reportUser({
+              reason: ReportReasons.mod,
+              message,
+              extra: "✅ Nudge",
+            });
+
+            const thread = await originalChannel.threads.create({
+              name: message.author.username,
+              autoArchiveDuration: 60,
+              // TODO: This won't work in servers that aren't at boost level 2
+              // Maybe could create a thread and ensure the "thread created" message is removed? honestly that's pretty invisible to anyone who isn't trawling through threads proactively
+              type: guild.features.includes("PRIVATE_THREADS")
+                ? "GUILD_PRIVATE_THREAD"
+                : "GUILD_PUBLIC_THREAD",
+              reason: "Private moderation thread",
+            });
+            const [{ moderator: modRoleId }] = await Promise.all([
+              fetchSettings(message.guild!, [SETTINGS.moderator]),
+              thread.members.add(message.author),
+            ]);
+            await thread.send(`The <@&${modRoleId}> team has determined that the following message is not okay in the community.
+
+This isn't a formal warning, but your message concerned the moderators enough that they felt it necessary to intervene. This message was sent by a bot, but all moderators can view this thread and are available to discuss what concerned them.
+
+  ${quoteAndEscape(message.content)}`);
+            return;
+          case resolutions.warning:
+            reportUser({
+              reason: ReportReasons.mod,
+              message,
+              extra: "✅ Warning",
+            });
+            message.reply(
+              `This message resulted in a formal warning from the moderators. Please review the community rules.`,
```

Comment:
Can we link the COC here?

Reply:
I want to do that, but this is already used in 3 servers so I need to add a means of configuring what url to use. Not a blocker for release I think, I want to improve onboarding for the bot and that's going to be a part